modules/pipeline/providers/actionmgr/db/client_test.go

```diff
@@ -0,0 +1,219 @@
+// Copyright (c) 2021 Terminus, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package db
+
+import (
+	"testing"
+	"time"
+
+	"github.com/alecthomas/assert"
+	"github.com/ghodss/yaml"
+	structpb "google.golang.org/protobuf/types/known/structpb"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/erda-project/erda-proto-go/core/pipeline/action/pb"
+)
+
+func TestPipelineAction_Convert(t *testing.T) {
+	type fields struct {
+		ID            string
+		TimeCreated   time.Time
+		TimeUpdated   time.Time
+		Name          string
+		Category      string
+		DisplayName   string
+		LogoUrl       string
+		Desc          string
+		Readme        string
+		Dice          string
+		Spec          string
+		VersionInfo   string
+		Location      string
+		IsPublic      bool
+		IsDefault     bool
+		SoftDeletedAt int64
+	}
+	type args struct {
+		yamlFormat bool
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    *pb.Action
+		wantErr bool
+	}{
+		{
+			name: "test no local",
+			fields: fields{
+				Spec: "name: release-fetch\nversion: \"1.0\"\ntype: action\nlogoUrl: http://terminus-paas.oss-cn-hangzhou.aliyuncs.com/paas-doc/2022/04/15/b23127cf-0ba7-48d1-b763-b8cd8c6e26d4.png\ncategory: deploy_management\ndesc: Fetch release by query\npublic: true\nsupportedErdaVersions:\n  - \">= 1.5\"\n\nparams:\n  - name: application_name\n    desc: The name of the application\n  - name: branch\n    desc: git branch to fetch first matching release\n\noutputs:\n  - name: release_id\n    desc: release id got from query\n\naccessibleAPIs:\n  - path: /api/applications\n    method: GET\n    schema: http\n\n  - path: /api/releases\n    method: GET\n    schema: http\n",
```

Comment:
1. use `` to format string spec to make testdata readable.

2. remove useless un-tested fields.